src/components/src/CartMain/cart.main.tsx

```diff
@@ -41,63 +45,361 @@ interface CartMainProps {
+    });
+  }
+
+  handleAddToList() {
+    const { itemData, listArray } = this.state;
+    const { onShowAlert } = this.props;
+    this.setState({ addToRequisitionListLoading: true });
+    if (listArray.length) {
+      const body: { [key: string]: any } = {};
+      body.items = itemData;
+      const promices = listArray.map(listUrl => cortexFetch(listUrl, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify(body),
+      }));
+      Promise.all(promices)
+        .then((res:any) => {
+          if (res[0].status === 201 || res[0].status === 200) {
+            this.handleModalClose();
+            this.fetchRequisitionListsData();
+            const isSuccess = true;
+            onShowAlert(intl.get('items-were-added-to-list-message'), isSuccess);
+          } else {
+            this.handleModalClose();
+            onShowAlert(intl.get('could-not-add-item-to-list-message'));
+          }
+        })
+        .catch((error) => {
+          onShowAlert(intl.get('could-not-add-item-to-list-message'));
+          // eslint-disable-next-line no-console
+          console.error(error.message);
+        });
+    } else {
+      this.setState({
+        openModal: false, addToRequisitionListLoading: false,
+      });
+    }
+  }
+
+  handleSelectList(list) {
+    const { listArray } = this.state;
+    const updatedListArray = listArray;
+    const index = listArray.indexOf(list.self.uri);
+    if (index !== -1) {
+      updatedListArray.splice(index, 1);
+    } else {
+      updatedListArray.push(list.self.uri);
+    }
+    this.setState({
+      listArray: updatedListArray,
+    });
+  }
+
+  handleRemove() {
+    const { onItemRemove } = this.props;
     onItemRemove();
   }
 
-  return (
-    <div className="cart-main-inner table-responsive">
-      {cartData._lineitems[0]._element.map(product => (
-        <CartLineItem
-          key={product._item[0]._code[0].code}
-          item={product}
-          handleQuantityChange={() => { handleQuantityChange(); }}
-          onRemove={handleRemove}
-          onConfiguratorAddToCart={handleConfiguratorAddToCart}
-          onMoveToCart={handleMoveToCart}
-          itemDetailLink={itemDetailLink}
-        />
-      ))}
-    </div>
-  );
-}
+  handleSaveList() {
+    const { listName, createRequisitionForm } = this.state;
+    if (listName.length !== 0) {
+      this.setState({ showReqListLoader: true });
+      cortexFetch(createRequisitionForm.self.uri, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify({
+          name: listName,
+        }),
+      })
+        .then((res) => {
+          if (res.status === 200 || res.status === 201) {
```

Comment:
Will an error still display as a banner if this fails?

Reply:
@shaunmaharaj I have updated this. If this fails, there will display message container with error and a modal window will close